### PR TITLE
ci: migrate to github actions

### PR DESCRIPTION
**Pull request by krampstudio**
- upgrade some dev tools 
 - migrate to GitHub Actions the following checks in CI:
   - conventional commits
   - unit test
   - coverage
   - eslint 

Eslint report job fails because the repo contains too many lint errors, it's not blocking for PRs but highlights the changes required.

### CODE REVIEW

File: .eslintrcignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+dist/
+node_modules/
+.nyc_ouput/
+coverage/
+lib/
+environment/
```
Comment (oatymart) [inline .eslintrcignore:6]: Nitpick: the standard name of the file is `.eslintignore`, without `rc` in the middle.

File: package.json [hunk 2]
```diff
@@ -15,12 +15,15 @@
...
     "coverage": "nyc report",
+    "coverage:clover": "nyc report --reporter=clover",
     "coverage:html": "nyc report --reporter=lcov && open-cli coverage/lcov-report/index.html",
     "build": "rollup --config ./build/rollup.config.js",
     "build:cov": "rollup --config ./build/rollup.config.js --environment COVERAGE",
     "build:dev": "NODE_ENV=development npm run build:watch",
     "build:watch": "rollup --config ./build/rollup.config.js --watch",
-    "lint": "eslint src/util src/core test",
+    "lint:src": "eslint -c .eslintrc.js src",
+    "lint:test": "eslint -c test/.eslintrc.js test",
```
Comment (oatymart) [inline package.json:25]: The file in `test/` doesn't seem to exist, so the command fails.

**Review by oatymart — CHANGES_REQUESTED** [review]
A much needed update! 🎉 
The CI reports are visible in the PR and are clearly working.
I ran some commands locally too (build, test, coverage, lint...) - and only 2 minor comments below.

File: .eslintrcignore (hunk 1 above)
Comment (krampstudio) [reply .eslintrcignore]: I've blindly copied it from the other repo. I'll have to fix it there as well

**Review by bziondik — APPROVED** [review]
Good job!

**Review by oatymart — APPROVED** [review]
Everything's good from my perspective!

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [ ] Documentation is updated according to changes (if applicable)

**Pull request merged**